tensorboard/plugins/pr_curve/summary_test.py

```diff
@@ -38,56 +39,240 @@ def setUp(self, *args, **kwargs):
     self.logdir = self.get_temp_dir()
     tf.reset_default_graph()
 
-  def testWeight1(self):
-    # Generate summaries for showing PR curves in TensorBoard.
-    with tf.Session() as sess:
-      summary.op(
-          tag='tag_bar',
-          labels=tf.constant([True, False, False, True, True, False]),
-          predictions=tf.constant([0.81, 0.8, 0.61, 0.6, 0.4, 0.2]),
-          num_thresholds=10)
-      merged_summary_op = tf.summary.merge_all()
-      foo_directory = os.path.join(self.logdir, 'foo')
-      writer = tf.summary.FileWriter(foo_directory, sess.graph)
-      writer.add_summary(sess.run(merged_summary_op), 1)
-      writer.close()
+    # Generate data.
+    pr_curve_demo.run_all(
+        logdir=self.logdir,
+        steps=3,
+        thresholds=5,
+        verbose=False)
 
     # Create a multiplexer for reading the data we just wrote.
-    multiplexer = event_multiplexer.EventMultiplexer()
-    multiplexer.AddRunsFromDirectory(self.logdir)
-    multiplexer.Reload()
+    self.multiplexer = event_multiplexer.EventMultiplexer()
+    self.multiplexer.AddRunsFromDirectory(self.logdir)
+    self.multiplexer.Reload()
 
+  def validateTensorEvent_(self, expected_step, expected_value, tensor_event):
+    """Checks that the values stored within a tensor are correct.
+    
+    Args:
+      expected_step: The expected step.
+      tensor_event: A TensorEvent named tuple.
+      expected_value: A nested python list of expected float32 values.
+    """
+    self.assertEqual(expected_step, tensor_event.step)
+    tensor_nd_array = tf.make_ndarray(tensor_event.tensor_proto)
+    np.testing.assert_allclose(
+        expected_value[5][2], tensor_nd_array[5][2], rtol=0, atol=1e-7)
```

Comment:
What are `[5][2]`?

Why not `np.testing.assert_allclose(expected_value, tensor_nd_array)`?

It seems like you're passing in a _lot_ of information to `validateTensorEvent_` and using hardly any of it.